### ISSUE TITLE
declaration-no-invalid

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#1085]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Good:
```css
body {
    width: 500px;
}
```

```css
body {
    width: 0;
}
```

Bad:
```css
body {
    width: 500;
}
```

**Comment by jeddy3** [comment]
This feels similar to https://github.com/stylelint/stylelint/issues/1003, where the proposal was to check the validity of certain property value pairs.

@davidtheclark Do you have a better feel for [`css-values`](https://github.com/ben-eb/css-values)'s applicability going forward in creating a `declaration-no-valid` rule, after your [discussion](https://github.com/ben-eb/css-values/issues/1) with Ben?

Unless you all feel differently, my guts still says we help move that project forward rather than maintaining our own list of properties that accept *length* values. 

Of course, anyone can experiment with such rules as plugins, but I'm hesitate to add rules to core that are likely to become redundant.

**Comment by evilebottnawi** [comment]
@jeddy3 i think this rule need in core, but pair with property and values maybe location in separate project(repo). Any rule check css syntax nice to see in the core, this will make one of the best styles linter.

**Comment by davidtheclark** [comment]
I have had little time to dedicate to css-values so far, and it's mostly been spent being kind of confused :) 

It is definitely the goal to produce functions that can validate declarations. So `declaration-no-invalid` can be made possible by it. @evilebottnawi if you want to push this forward it might be a really good idea to take a look at css-values and see if you can help out there. Hopefully I will get more time to do that soon.

**Comment by evilebottnawi** [comment]
@davidtheclark i think we should rename `declaration-no-invalid` and set `help wanted`, `rule`. Most likely I will have time to start making this rule, in the future we will be able to bring it to perfection.

**Comment by jeddy3** [comment]
> i think we should rename declaration-no-invalid and set help wanted, rule

I don't have a problem with that. If only if this issue will then just act as a pointer towards [css-values](https://github.com/ben-eb/css-values).

> Most likely I will have time to start making this rule

It's great that you think you'll have time to help out at [css-values](https://github.com/ben-eb/css-values)! :) 

> in the future we will be able to bring it to perfection.

I think we'd only want to merge this rule into core *once it is perfected and finished* i.e. it correctly checks the validity of *every declaration* (which sounds like it will require [css-values](https://github.com/ben-eb/css-values)). Otherwise, we get into a situation that every addition to the rule is a *breaking change* as it'll likely flag up more warnings in users CIs. Which is why, once a rule is released, we tend to only add changes behind options.

**Comment by jeddy3** [comment]
@evilebottnawi I'm renaming the issue. Please only open a Pull Request *once the rule can be perfected and finished* i.e. after [css-values](https://github.com/ben-eb/css-values) is finished, or we'll have a lingering PR for a while!

**Comment by jeddy3** [comment]
Bumping to plugin.

**Comment by evilebottnawi** [comment]
@jeddy3 i think be good add this rule in core, not now, later.

**Comment by evilebottnawi** [comment]
@jeddy3 it is be a great rule. For solutions we just need to have a repository with the actual css specifications in the required format.

**Comment by evilebottnawi** [comment]
@jeddy3 `unit-no-unknown` also never be `finished`, specs can add new units and we should add units for check also, if use this strategy with all rules we leave only stylistics rule and it is not good.

**Comment by jeddy3** [comment]
> i think be good add this rule in core, not now, later.

[That's the plan](https://github.com/stylelint/stylelint/issues/1238#issuecomment-219237306).

>  Maybe for that reason this should be a plugin with 0.x versions until it stabilizes, and then we pull it back into core?

**Comment by evilebottnawi** [comment]
@jeddy3 good, we can always pull it into the core when it is stable :tada:

**Comment by ben-eb** [comment]
Please ping me if you have questions, I don't watch this repo! 😄
